packages/shared/src/hooks/useDebounceFn.ts

```diff
@@ -0,0 +1,56 @@
+import { useCallback, useEffect, useRef, useMemo } from 'react';
+
+export type StartFn<T, R = void> = (params?: T) => R;
+export type CancelEvent = () => void;
+
+export default function useDebounceFn<T = unknown>(
```

Comment:
This is just the old file renamed?

Reply:
Non blocking but to avoid this many changes the new hook could have been `useDebouncedValue`.

Reply:
If you prefer I can do that?

Reply:
Since you already did the work I am ok with this as well 👍